server.cpp

```diff
@@ -53,7 +55,62 @@ void write_log(const std::string& outfile, const std::string& log_message)
     }
 }
 
-void write_response(tcp::socket& socket, const Request& request, const std::string& date)
+std::string make_message(DIR *dir, const std::string& path, const std::string& date)
+{
+        std::string file_name;
+        std::string line;
+
+        for (struct dirent *entry = readdir(dir); entry != NULL; entry = readdir(dir))
+        {
+            if (strcmp(".", entry->d_name) && strcmp("..", entry->d_name))
+            {
+                file_name = entry->d_name;
+
+                struct stat st;
+                if (stat((path + "/" + file_name).c_str(), &st) < 0)
+                {
+                    line = line + "<tr><td>" + file_name + "</td><td>?</td><td>?</td></tr>";
+                }
+                else
+                {
+                    std::string file_date = asctime(localtime(&st.st_ctime));
+                    line = line + "<tr><td><p><a href=\"" + file_name + "\">" + file_name + "</a></p></td><td>" + std::to_string(st.st_size) + "</td><td>" + file_date + "</td></tr>";
+                }
+            }
+        }
+        closedir(dir);
+
+        const std::string table_html ="<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.01//EN\" \"http://www.w3.org/TR/html4/strict.dtd\"> \
+            <html> \
+            <body> \
+            <table border=\"1\" cellspacing=\"0\" cellpadding=\"5\"> \
+            <tr><td>File name</td><td>File size</td><td>Last modified date</td></tr>" + line + " \
+            </table> \
+            </body> \
+            </html>";
+
+        return "HTTP/1.1 200 OK\r\nHost: localhost\r\nContent-Type: text/html; charset=utf-8\r\n\r\n" + date + "\r\n" + table_html;
+}
+
+void write_file(tcp::socket& socket, const std::string& request_path_file, const std::string& path, error_code ignored_error)
+{
+    std::ifstream fin(path + "/" + request_path_file);
+
+    char buff[1024] = {0};
+    while (!fin.eof())
+    {
+        fin.read(buff, 1024);
+
+        std::streamsize len = fin.gcount();
+        if (len > 0)
+        {
+            std::string msg(buff, len);
```

Comment:
Why do you need this string? Why can't you create `ba::buffer` directly from the `buff`?